sdk/servicebus/Azure.Messaging.ServiceBus/tests/Processor/SessionProcessorLiveTests.cs

```diff
@@ -1389,5 +1397,74 @@ async Task ProcessMessage(ProcessSessionMessageEventArgs args)
                 Assert.AreEqual(1, messageCt);
             }
         }
+
+        [Test]
+        [TestCase(10, 10)]
+        [TestCase(10, 5)]
+        [TestCase(10, 20)]
+        public async Task MaxCallsPerSessionRespected(int numSessions, int maxConcurrentSessions)
+        {
+            await using (var scope = await ServiceBusScope.CreateWithQueue(
+                enablePartitioning: false,
+                enableSession: true))
+            {
+                await using var client = GetClient();
+                ServiceBusSender sender = client.CreateSender(scope.QueueName);
+
+                ConcurrentDictionary<string, bool> sessions = new ConcurrentDictionary<string, bool>();
+                int messagesPerSession = 10;
+                int totalMessages = messagesPerSession * numSessions;
+                for (int i = 0; i < numSessions; i++)
+                {
+                    await sender.SendMessagesAsync(GetMessages(messagesPerSession, Guid.NewGuid().ToString()));
+                }
+
+                TaskCompletionSource<bool> tcs = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
+                int messageCt = 0;
+
+                var options = new ServiceBusSessionProcessorOptions
+                {
+                    MaxConcurrentSessions = maxConcurrentSessions,
+                    MaxConcurrentCallsPerSession = 1,
```

Comment:
I would use something other than 1

Reply:
This test is meant to verify the default behavior, but I can parameterize this and update the dictionary to store off the count of concurrent processing.